src/native/corehost/runtime_config.cpp

```diff
@@ -334,9 +334,9 @@ bool runtime_config_t::ensure_dev_config_parsed()
     trace::verbose(_X("Attempting to read dev runtime config: %s"), m_dev_path.c_str());
 
     pal::string_t retval;
-    if (!pal::file_exists(m_dev_path))
+    if (m_dev_path.empty() || !pal::realpath(&m_dev_path, true))
```

Comment:
> m_dev_path.empty()

Does `realpath` not handle that? I would have expected it to treat an empty string as not existing / having a real path and return false?

Reply:
On Windows we would return true on an empty path
https://github.com/dotnet/runtime/blob/eec31225f730368917dfd610352ae46543001cd4/src/native/corehost/hostmisc/pal.windows.cpp#L648-L656

`file_exists` does the same check on Windows:
https://github.com/dotnet/runtime/blob/eec31225f730368917dfd610352ae46543001cd4/src/native/corehost/hostmisc/pal.windows.cpp#L716-L719

 Maybe this check should be moved to `realpath`?

Reply:
Ha, but then result is the same... so yeah, I think it's handled already.

Reply:
I took a second look at this. Based on the comment on `realpath`: `// Return if path is valid and file exists, return true and adjust path as appropriate.` I think the above should not return true as it does today when passing an empty path. I changed the if-condition, PTAL.